src/Interpreters/InterpreterInsertQuery.cpp

```diff
@@ -175,7 +176,10 @@ StoragePtr InterpreterInsertQuery::getTable(ASTInsertQuery & query)
         query.table_id = current_context->resolveStorageID(local_table_id);
     }
 
-    return DatabaseCatalog::instance().getTable(query.table_id, current_context);
+    auto table = DatabaseCatalog::instance().getTable(query.table_id, current_context);
```

Comment:
Is it possible to return the ref table inside `getTable` interface. there are too much places to add in `src/Interpreters`. For example, `truncate` and system command looks not work.

Reply:
I have considered to override all the methods of interface `IStorage`, but I think it makes things more complicated. Some drawbacks:

1. Every call will lead to a retrieve inside `DatabaseCatalolg` for ref table.
2. Every time we modify `IStorage` we should consider `StorageAlias`. We may forget to update `StorageAlias` when we add a virtual method with default implementation.
3. It's difficult to figure out which method should be overriden.

Reply:
> I think we can add a bool member in Context, like `return_ref_table_for_alias_storage`. and the default is true, for some special query like `drop` , we set it as false.
> 
> The code will still be clean, because this kind of special query is few.

LGTM